Add filter class to dask and do the tests for it

xuzifan08 opened the pull request:
Added filter class to dask.py and four tests associated with it.

CJ-Wright commented on streamz/tests/test_dask.py:
```diff
@@ -131,6 +131,67 @@ def test_buffer(c, s, a, b):
 
 
+@pytest.mark.slow
+def test_filter(backend):
```
You need to remove the `backend` arg here, and in the `scatter` statement. Streamz only has a dask backend at the moment.

xuzifan-lab replied:
Should I remove the backend for all the four tests?

xuzifan08 replied:
Should I remove the backend for all the four tests I have added?

CJ-Wright replied:
yes please

xuzifan08 replied:
Already changed the four tests but the build is still failed

CJ-Wright commented on streamz/dask.py:
```diff
@@ -140,6 +157,24 @@ def update(self, x, who=None):
 
 
+@DaskStream.register_api()
+class filter(DaskStream):
```
I think you also need the modifications to the gather and other nodes as well.

xuzifan08 replied:
I made changes to gather already. I also compared other nodes. There is a slightly difference between the dask.starmap and parallel.starmap. Do I need to change that one?

xuzifan08 replied:
I made changes to gather already. I also compared other nodes. There is a slightly difference between the dask.starmap and parallel.starmap. Do I need to change that one?

xuzifan08 commented:
Hi, @martindurant @mrocklin, this is Zifan. I'm working on this branch with @CJ-Wright . 

We added class filter() to dask and modified class gather() according to parallel.py (including the associated tests). However, we ran into an issue which causes the test test_integration_from_stream fail in the test_dataframe.py. The test passes on the Stream but fails on Daskstream. After troubleshooting for a while, we guess there is something wrong with the process of serializing the code to different dask clusters, but we are not sure what the specific issue is. 

Can we pick your brains about how we can resolve it? Thanks!

martindurant commented:
Sorry for not following up here - are you still interested in this, @xuzifan08 ?
